Code-map.md

```diff
@@ -0,0 +1,101 @@
+# A map to the Vespa code base
+
+You want to get familiar with the Vespa code base but don't know where to start?
+
+Vespa consists of about 1.7 million lines of code, about equal parts Java and C++.
+Since it it's mostly written by a team of developers selected for their ability 
+to do this kind of thing unusually well, who have been given time to dedicate 
+themselves to it for a long time, it is mostly easily to work with. However, one 
+thing we haven't done is to create a module structure friendly to newcomers - the code
+simply organized in a flat structure of about 150 modules. 
+
+This document aims to provide a map from the
+[functional elements](https://docs.vespa.ai/documentation/overview.html)
+of Vespa to the most important modules in the flat module structure in the
+[code base on GitHub](https://github.com/vespa-engine/vespa).
+
+It covers the modules you are most likely to encounter as a developer. 
+The rest are either small and needed for technical reasons or doing one thing 
+which should be self-explanatory, or implementing the cloud service run by the 
+Vespa team which we don't expect anybody else to run and therefor be interested 
+in changing.
+
+
+## The stateless container
+
+When a request is made to Vespa it first enters some stateless container cluster,
+called jDisc. This consists of:
+
+- a __jDisc core__ layer which provides a model of a running application, general protocol-independent request-response handling, with various protocol implementations,
+- a __jDisc container__ layer providing component management, configuration and similar.
+- a __search middleware__ layer containing query/result API's, query execution logic etc.
+- API's and modules for writing and processing document operations.
+
+The stateless container is implemented in Java.
+
+jDisc core modules:
+
+- [jdisc_core](https://github.com/vespa-engine/vespa/tree/master/jdisc_core) - The core jDisc functionality
+- [jdisc_http_service](https://github.com/vespa-engine/vespa/tree/master/jdisc_http_service) - HTTP connector for jDisc, implemented using Jetty.
+
+jDisc container modules, layered on jDisc core:
+
+- [container-disc](https://github.com/vespa-engine/vespa/tree/master/container-disc) - Integration between the jDisc container and jDisc core layers.
+- [container-core](https://github.com/vespa-engine/vespa/tree/master/container-core) - Core jDisc container functionality: Metrics, OSGi integration for component bundles, etc.
+- [component](https://github.com/vespa-engine/vespa/tree/master/component) - the component model. Components (in Java) will implement or subclass a type for this module.
```

Comment:
Start with upper case 'T' in 'the component...' for consistency? Applies for the next three items as well

Reply:
Done (lowercase everywhere)